### PR TITLE
run `ineffassign` for all code

### LINKED ISSUE #12543: Enable Ineffassign To Run for Tests
# 💎 Issue

### Background

Currently we have the `ineffassign` static analyzer running on our production code for each build, however this is disabled for our test code. This leads to unhandled errors in our tests, which could potentially hide relevant test failures. 

### Description

We will need to fix our `ineffassign` analyzer [here](https://github.com/prysmaticlabs/prysm/tree/develop/tools/analyzers/ineffassign) to be able to run on test code. Currently this is not the case.

Audit Finding: TOB-PRYSM-1

### PR DESCRIPTION
**Pull request by anukul**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

Updates bazel nogo config to run the `ineffassign` static analyzer for all go files

**Which issues(s) does this PR fix?**

Fixes #12543

**Other notes for review**

Confirmed that `ineffassign` does run for tests, but the `only_files` filter was excluding a lot of files.

### CODE REVIEW

**Comment by anukul** [comment]
There are 2 more analyzers configured on the same subset of source files - `cryptorand` and `properpermissions`.

Should they be updated similarly?

File: testing/endtoend/components/lighthouse_validator.go
Comment (nisdas) [inline testing/endtoend/components/lighthouse_validator.go:1]: Revert these changes here, they don't have anything to do with ineffassign

Comment (anukul) [reply testing/endtoend/components/lighthouse_validator.go:1]: I noticed that `beaconRPCPort` was reported by `ineffassign` and it doesn't seem to be used after assignment.

Comment (nisdas) [reply testing/endtoend/components/lighthouse_validator.go:1]: Then you can just remove `beaconRPCPort`, there are other changes that have been added along with it

Comment (anukul) [reply testing/endtoend/components/lighthouse_validator.go:1]: Updated.
Apologies for the mixup, I didn't realize those changes would be redundant.

**Review by nisdas — DISMISSED** [review]
Approved, thanks for picking this up !

**Comment by nisdas** [comment]
```


compilepkg: nogo: errors found by nogo during build-time code analysis:
--
  | testing/endtoend/component_handler_test.go:257:2: ineffectual assignment to "msg" (ineffassign)
  | testing/endtoend/endtoend_test.go:519:3: ineffectual assignment to "index" (ineffassign)

```
There seem to be a few failures here

File: testing/endtoend/components/lighthouse_validator.go
Comment (anukul) [reply testing/endtoend/components/lighthouse_validator.go:1]: Worth noting that `httpPort+index` only works as long as `e2e.TestParams.BeaconNodeCount` `==` `e2e.TestParams.LighthouseBeaconNodeCount`, which is currently [true](https://github.com/prysmaticlabs/prysm/blob/develop/testing/endtoend/params/params.go#L114-L118) but will implicitly break things if this constraint changes in future. 

While I cannot say for certain, my guess is that the deleted `beaconRPCPort` code was meant to set `beaconGatewayPort` in a way that handles the case when this constraint is not true.

**Review by nisdas — APPROVED** [review]

**Comment by nisdas** [comment]
> There are 2 more analyzers configured on the same subset of source files - [cryptorand](https://github.com/prysmaticlabs/prysm/blob/develop/nogo_config.json#L101-L106) and [properpermissions](https://github.com/prysmaticlabs/prysm/blob/develop/nogo_config.json#L158-L163).

> Should they be updated similarly?

I don't think that is necessary, we don't need these analyzers to run on test files, just on production code. You can make a follow up PR for these 2 analyzers where they are running on all our source production code ( excluding test files)

**Review by prestonvanloon — APPROVED** [review]
Nice work!
